_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,354 @@
+Image from the `DockerFile` included in the example code. This `DockerFile` 
+is fairly basic:
+
+```bash
+FROM php:7
+RUN apt-get update -y && apt-get install -y openssl zip unzip git
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+RUN docker-php-ext-install pdo mbstring
+WORKDIR /app
+COPY . /app
+RUN composer install
+
+CMD php artisan key:generate
+CMD php artisan serve --host=0.0.0.0 --port=8181
+EXPOSE 8181
+``` 
+
+It extends a PHP base image, installs some system dependencies including 
+composer and the standard PHP extensions required by Laravel. It will then
+ copy the application files to a working directory and install the application's 
+  dependencies via composer. It also runs a couple of `artisan` commands, 
+  last of which serves the application using the built-in PHP web server. At the
+   end, it exposes port `8181` to the deployment.
+  
+To build the local Docker image:
+
+```bash
+cd /to/your/project/directory
+docker build -t yourname/laravel-kubernetes-demo .
+```
+
+__Deploying the image__
+
+Now that the application's image is built and is available for deploying it 
+locally, let's go ahead and deploy it.
+
+I always start with making sure that Kubectl is in the right context. In our 
+case, the context is Minikube. We can simply switch context as follows:
+
+```bash
+kubectl config use-context minikube
+```
+
+then we can deploy the container image:
+
+```bash
+kubectl run laravel-kubernetes-demo --image=yourname/laravel-kubernetes-demo 
+--port=8181 --image-pull-policy=IfNotPresent
+```
+
+The above command tells Kubectl to run our demo application from the Docker 
+image, whilst making port 8181 available for listening. The last parameter of 
+the command simply asks Kubectl to not pull the image from a registry such as
+ Docker Hub if it exists locally which in our case it does. Do note that you 
+ still need to be logged on to Docker so that Kubectl can check if the image 
+ is up to date.
+ 
+We can check that a Pod is created for our application by running:
+
+```bash
+kubectl get pods
+```
+
+which should return a similar output to:
+
+```bash
+NAME                                       READY     STATUS    RESTARTS   AGE
+laravel-kubernetes-demo-7dbb9d6b48-q54wp   1/1       Running   0          18m
+```
+
+We can also use the Minikube GUI dashboard to monitor our cluster. The GUI 
+also helps with visualising most of the discussed concepts. To view the 
+dashboard, simply run the following:
+
+```bash
+minikube dashboard
+```
+
+or to acquire the dashboard's URL address:
+
+```bash
+minikube dashboard --url=true
+```
+
+which will return the IP address and the port number of the dashboard.
+
+__Exposing a service__
+
+So far we have created a deployment which is running our application's 
+container. With the container running, we need to expose the PHP's built-in web 
+server as a service.
```

Comment:
I think it's important to explain why you need a service.
I suggest adding something like this:

So far we have created a deployment which is running our application's container. A pod running in the cluster has a dynamic IP. If we route the traffic directly to it using the IP, we may need to update the routing table every time the pod is restarted. In fact, every time we deploy or restart the container, a new IP is assigned to the pod. To avoid managing IP addresses manually, we're going to use a Service. The service as an IP and act as a load balancer for a set of pods. So even if the IP address of a pod change, the service is always pointing to it. And since the Service has always the same IP, we don't need to update anything.

Reply:
I agree, now. Since I'm removing the block explaining the terms which said more or less the same thing.